testcases/misc_tests/p11sak_test.sh

```diff
@@ -0,0 +1,419 @@
+#!/bin/bash
+#
+# COPYRIGHT (c) International Business Machines Corp. 2020
+#
+# This program is provided under the terms of the Common Public License,
+# version 1.0 (CPL-1.0). Any use, reproduction or distribution for this software
+# constitutes recipient's acceptance of CPL-1.0 terms which can be found
+# in the file LICENSE file or at https://opensource.org/licenses/cpl1.0.php
+
+# - The PKCS11_USER_PIN environment variable must hold the user pin.
+# - The SLOT environment variable must hold the slot id of the token under test.
+#
+# sudo -E ./p11sak_test.sh
+
+
+
+echo "** Now executing 'p11sak_test.sh'"
+
+
+# tmp files
+
+P11SAK_DES_PRE=p11sak-des-pre.out
+P11SAK_DES_POST=p11sak-des-post.out
+P11SAK_3DES_PRE=p11sak-3des-pre.out
+P11SAK_3DES_POST=p11sak-3des-post.out
+P11SAK_AES_PRE=p11sak-aes-pre.out
+P11SAK_AES_POST=p11sak-aes-post.out
+P11SAK_RSA_PRE=p11sak-rsa-pre.out
+P11SAK_RSA_POST=p11sak-rsa-post.out
+P11SAK_EC_PRE=p11sak-ec-pre.out
+P11SAK_EC_POST=p11sak-ec-post.out
+
+
+
+echo "** Now generating keys - 'p11sak_test.sh'"
+
+
+# generate objects
+
+# des
+p11sak generate-key des --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-des
```

Comment:
How would the CI know if such a test step fails? 
I don't see any return code checking, etc.... The script would just continue to execute. Is there a subsequent process that scans the output for errors?

Reply:
See below